web/react/components/post.jsx

```diff
@@ -223,6 +244,10 @@ export default class Post extends React.Component {
                                 posts={posts}
                                 handleCommentClick={this.handleCommentClick}
                                 retryPost={this.retryPost}
+                                setHasEmbedState={this.setHasEmbedState}
+                                toggleEmbedVisibility={this.toggleEmbedVisibility}
+                                hasEmbed={this.state.hasEmbed}
+                                embedVisible={this.state.embedVisible}
```

Comment:
The postHeader previously displayed the expand icon, so it needed this data passed through postInfo. If the final placement of the icon is not going to be in the header, this can certainly be removed. As for post vs. postBody, that was again related to the change- I simply already had the logic in place. That too can be moved, unless you foresee a reason the rest of the post might need information about the embed preview visibility in the future. Does that seem likely?

-Alan

On Feb 22, 2016, 5:37 AM -0800, Joram Wilander<notifications@github.com>, wrote:
>  
> Inweb/react/components/post.jsx(https://github.com/mattermost/platform/pull/2027#discussion_r53624747):
>  
> >@@ -223,6 +244,10 @@ export default class Post extends React.Component {>posts={posts}>handleCommentClick={this.handleCommentClick}>retryPost={this.retryPost}>+                setHasEmbedState={this.setHasEmbedState}>+                toggleEmbedVisibility={this.toggleEmbedVisibility}>+                hasEmbed={this.state.hasEmbed}>+                embedVisible={this.state.embedVisible}
>  
> It looks like the PostBody is the only component that makes use of the embed visibility. Do you think it's worth having the extra logic to keep the state stored in the Post itself rather just encapsulating it all within PostBody?
>  
>  
> —
> Reply to this email directly orview it on GitHub(https://github.com/mattermost/platform/pull/2027/files#r53624747).
>  
>  
>